# snapTotal-seq
This repository provides the data analysis pipeline of single cell RNA-seq data generated by snapTotal-seq.

Instructions:
The libraries prepared by using snapTotal-seq can be sequenced using either customized primers or Illumina standard sequencing primers. If the library is sequenced using customized primers, the fastq files of each single cell can be demultiplexed by using Illumina standard demultiplex pipeline. The users can directly proceed with mapping (01_snapTotal_mapping.sh) and UMI counting (02_split_exon_intron_reads_customized_run.sh). If the library is sequenced using Illumina standard sequencing primer, the library will be demultiplexed into several sub-libraries based on i5 index using Illumina standard demultiplex pipeline. 

Requirements:
1. STAR (v2.5.3a)
2. Cutadapt (v3.4)
3. Python (v3.8.8)
4. Pysam (v0.16.0.1)
5. Htseq-count (v0.13.5)
6. Samtools (v1.12)
7. seqtk (v1.3)
8. Pandas (v1.2.4)
The versions we used are listed in parentheses. 
